lib/Doctrine/Common/Collections/AbstractLazyCollection.php

```diff
@@ -35,6 +38,7 @@ abstract class AbstractLazyCollection implements Collection
     public function count()
     {
         $this->initialize();
+        assert($this->collection !== null);
```

Comment:
Do we really need to C&P this everywhere? Calling the `initialize()` method already makes sure that the collection is initialized.

Reply:
It's for static analysis.

Adding 
```
@psalm-assert Collection<TKey,T> $this->collection
```
works for psalm, but phpstan doesn't understand this tag yet.

So it reports
```
 Cannot call method offsetUnset() on Doctrine\Common\Collections\Collection<TKey of (int|string), T>|null.
```
But I can add the pattern to the ignoreError section, until it's supported by phpstan (which should happen soon).

Reply:
I updated the PR

Reply:
That works for me, thanks.